x-pack/osquerybeat/internal/pub/publisher_test.go

```diff
@@ -112,3 +113,92 @@ func TestHitToEvent(t *testing.T) {
 		}
 	}
 }
+
+func TestActionResultToEvent(t *testing.T) {
+
+	tests := []struct {
+		name     string
+		req, res map[string]interface{}
+		want     map[string]interface{}
+	}{
+		{
+			name: "successful",
+			req: toMap(t, `{
+				"data": {
+					"id": "a72d65d8-200a-4b43-8dbd-7bc0e9ce8e65",
+					"query": "select * from osquery_info"
+				},
+				"id": "5c433f88-ab0d-41e2-af76-6ff16ae3ced8",
+				"input_type": "osquery",
+				"type": "INPUT_ACTION"
+			}`),
+			res: toMap(t, `{
+				"completed_at": "2024-04-18T19:39:39.740162Z",
+				"count": 1,
+				"started_at": "2024-04-18T19:39:39.532125Z"
+			} `),
+			// "agent_id": "bf3d6036-2260-4bbf-94a3-5ccce0d75d9e",
+			want: toMap(t, `{
+				"completed_at": "2024-04-18T19:39:39.740162Z",
+				"action_response": {
+					"osquery": {
+						"count": 1
+					}
+				},
+				"action_id": "5c433f88-ab0d-41e2-af76-6ff16ae3ced8",
+				"started_at": "2024-04-18T19:39:39.532125Z",
+				"action_input_type": "osquery",
+				"action_data": {
+					"id": "a72d65d8-200a-4b43-8dbd-7bc0e9ce8e65",
+					"query": "select * from osquery_info"
+				}
+			}`),
+		},
+		{
+			name: "error",
+			req: toMap(t, `{
+				"data": {
+					"id": "08995ee8-5182-423e-9527-552736411010",
+					"query": "select * from osquery_foo"
+				},
+				"id": "70539d80-4082-41e9-aff4-fbb877dd752b",
+				"input_type": "osquery",
+				"type": "INPUT_ACTION"
+			}`),
+			res: toMap(t, `{
+				"completed_at": "2024-04-20T14:56:34.87195Z",
+				"error": "query failed, code: 1, message: no such table: osquery_foo",
+				"started_at": "2024-04-20T14:56:34.87195Z"
+			}`),
+			// "agent_id": "bf3d6036-2260-4bbf-94a3-5ccce0d75d9e",
+			want: toMap(t, `{
+				"completed_at": "2024-04-20T14:56:34.87195Z",
+				"action_id": "70539d80-4082-41e9-aff4-fbb877dd752b",
+				"started_at": "2024-04-20T14:56:34.87195Z",
+				"action_input_type": "osquery",
+				"error": "query failed, code: 1, message: no such table: osquery_foo",
+				"action_data": {
+				  "id": "08995ee8-5182-423e-9527-552736411010",
+				  "query": "select * from osquery_foo"
+				}
+			  }`),
+		},
+	}
+
+	for _, tc := range tests {
```

Comment:
If you use `T.Run()` here, it will have some advantages, each test case would be run and return results separately.

Reply:
Ooops, sorry, saw the approval, merged, but missed these comments. Will open a follow up PR to address this.